### DISCUSSION TITLE
is there a hook to start save from userscript

### DISCUSSION THREAD

**Discussion by saksaikot** [gildas-lormeau/SingleFile#750] [Q&A]
i'll be very help full if saveFile can be triggered by userscript, so then custom automatic save will be possible

**Comment by gildas-lormeau** [comment]
It's not possible because it's not really possible to implement this feature in a secure way. The problem is that a webpage could call SingleFile too. To my knowledge, there is no mechanism that allows a user script to communicate with an extension without the page also being able to do so.

Note that automatic save is already possible with SingleFile though.

**Reply by saksaikot** [reply]
Thanks, automatic save is working, only problem, if I use before hook then the frame content is not loading, but without hook frame content loaded successfully.

**Reply by kwhitefoot** [reply]
Perhaps it could be allowed by having a whitelist of pages that were allowed to do it?

**Comment by Lokawn** [comment]
@gildas-lormeau Could you please elaborate on how to use [this](https://github.com/gildas-lormeau/SingleFile/issues/1186#issuecomment-1646838753) in userscript ?

**Reply by gildas-lormeau** [reply]
Unfortunately, it looks like user scripts do not offer any API to call [`runtime.sendMessage()`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage).
